app/routes/context_window.py

```
@@ -0,0 +1,940 @@
+        logger.error(
+            "trigger_async_compaction_failed",
+            conversation_id=conversation_id,
+            error=str(e)
+        )
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.get("/{conversation_id}/compact/progress", response_model=CompactionStatusResponse)
+async def get_compaction_progress(
+    conversation_id: str,
+    user_id: str = Depends(get_current_user)
```

Ruff (0.14.14)

[warning] Unused function argument: `user_id`

(ARG001)